### PR TITLE
Upgrade to 2.11.8

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd.

**Comment by nicolasstucki** [comment]
This PR conflicts with #2289, the second one being merged will need a rebase on the first one.

File: scalalib/overrides-2.11.8/scala/concurrent/impl/AbstractPromise.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package scala.concurrent.impl
+
+import java.util.concurrent.atomic.AtomicReference
+
+@Deprecated // Since 2.11.8. Extend java.util.concurrent.atomic.AtomicReference instead.
+abstract class AbstractPromise extends AtomicReference[AnyRef] {
+  protected final def updateState(oldState: AnyRef, newState: AnyRef): Boolean =
+      compareAndSet(oldState, newState)
```
Comment (sjrd) [inline scalalib/overrides-2.11.8/scala/concurrent/impl/AbstractPromise.scala]: Indentation issue.

File: ci/checksizes.sh [hunk 2]
```diff
@@ -39,7 +39,7 @@ case $FULLVER in
     REVERSI_PREOPT_GZ_EXPECTEDSIZE=71000
     REVERSI_OPT_GZ_EXPECTEDSIZE=32000
     ;;
-  2.11.7)
+  2.11.8)
     REVERSI_PREOPT_EXPECTEDSIZE=535000
     REVERSI_OPT_EXPECTEDSIZE=124000
```
Comment (sjrd) [inline ci/checksizes.sh]: We should check whether this improved. I hope so; I had contributed a PR to 2.11.8 that was supposed to remove the dependency of `Vector` on `System.out.println`.

Comment (nicolasstucki) [inline ci/checksizes.sh]: These where the results:
```
Checksizes: Scala version: 2.11.8
Reversi preopt size = 523770 (expected 535000)
Reversi opt size = 120502 (expected 124000)
Reversi preopt gzip size = 70242 (expected 73000)
Reversi opt gzip size = 31679 (expected 33000)
```

Comment (gzm0) [inline ci/checksizes.sh]: Nice!

Comment (nicolasstucki) [inline ci/checksizes.sh]: Should I tighten them a bit?

Comment (gzm0) [inline ci/checksizes.sh]: Of course! Tighten them screws!!! :)

**Comment by sjrd** [comment]
Nightly: https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/3791/
Conditional LGTM on nightly passing.

**Comment by sjrd** [comment]
LGTM, nightly passed.

**Pull request merged**